api/bundle/client.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package bundle provides access to the bundle api facade.
+// This facade contains api calls that are specific to bundles.
+
+package bundle
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/apiserver/params"
+)
+
+var logger = loggo.GetLogger("juju.api.bundle")
+
+// Client allows access to the bundle API end point.
+type Client struct {
+	base.ClientFacade
+	facade base.FacadeCaller
+}
+
+// NewClient creates a new client for accessing the bundle api.
+func NewClient(st base.APICallCloser) *Client {
+	frontend, backend := base.NewClientFacade(st, "Bundle")
+	return &Client{
+		ClientFacade: frontend,
+		facade:       backend}
+}
+
+// ExportBundle exports the current model configuration.
+func (c *Client) ExportBundle() (string, error) {
+	var result params.StringResult
+	if bestVer := c.BestAPIVersion(); bestVer < 2 {
+		return "", errors.Errorf("command not supported on v%d", bestVer)
+	}
+
+	if err := c.facade.FacadeCall("ExportBundle", nil, &result); err != nil {
+		errors.Annotate(result.Error, "export failed")
```

Comment:
No we do not want to annotate result.Error here. In fact we do not even want to touch here. 
In fact, there is no guarantee that result is valid since you got an error from the call.

Reply:
I have removed.

Reply:
@anastasiamac We can discuss this tomorrow. I do understand that we have 2 diff errors. 
But in this scenario from server function call ExportBundle() when the function returns error(ie non-nil) the result.Error is always nil. And, in case of  err is nil, there are no scenarios in that function that sets result.Error like for example in credentialValidator.go @@Line77 and @@78.

So what i was trying to point out it for this function ExportBundle(0 from sever side we never come across a state where err=nil  but result.Error="having some value".